postprocess/patch_operator_deployment.jsonnet

```
@@ -0,0 +1,29 @@
+local com = import 'lib/commodore.libjsonnet';
+local inv = com.inventory();
+
+local deployment_file = std.extVar('output_path') + '/deployment.yaml';
+local deployment = com.yaml_load(deployment_file);
+local on_openshift4 = inv.parameters.facts.distribution;
+
+{
+  deployment: deployment {
+    spec+: {
+      template+: {
+        spec+: {
+          containers: [
+            c {
+              securityContext+: {
+                runAsUser:
+                  if on_openshift4 == 'openshift4' then
+                    null
+                  else
+                    c.securityContext.runAsUser,
+              },
+            }
+            for c in deployment.spec.template.spec.containers
+          ],
+        },
+      },
+    },
+  },
+}
```

I would make this more generic, so any occurrences of `runAsUser` are patched, instead of having to keep a list of files to patch here.

Something like:

```suggestion
local helmchart_dir = std.extVar('output_path');
local on_openshift4 = inv.parameters.facts.distribution;

local removeRunAsUser(obj) =
  if obj.kind == 'Deployment' then
    obj {
     spec+: {
      template+: {
        spec+: {
          containers: [
            c {
              securityContext+: {
                runAsUser: null
              },
            }
            for c in deployment.spec.template.spec.containers
          ],
        },
      },
    }
  else
    obj;

if on_openshift4 then
  com.fixupDir(helmchart_dir, removeRunAsUser)
else 
  {}
```

See https://syn.tools/commodore/reference/commodore-libjsonnet.html#_fixupdirdir_fixupfn for details how `com.fixupDir()` works.

---

I implemented almost the same logic while fixing the review :)